### PR TITLE
added matcher class and functions

### PR DESCRIPTION
**Pull request by shashank-boyapally**
## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Added Matcher class and functions. Finding uuids pertaining to a metadata, cross-referencing with the jobs in ripsaw-kube-burner and saving them to csv.

Requires further changes including creating a class for elastic service and configs

## Related Tickets & Documents

- Related Issue #
- Closes #

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please describe the System Under Test.
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,2 @@
 # py-commons
-Common Python Library 
+This common python library is dedicated to metadata matching and finding regressions.
```
Comment (jtaleric) [inline README.md]: This line might be more appropriate in the fmatch directory.

**Review by jtaleric — COMMENTED** [review]
lets also remove the .DS_Store (Mac :P)

File: README.md (hunk 1 above)
Comment (shashank-boyapally) [reply README.md]: Thank you for the suggestion Joe. I'll add the readme to fmatch.

File: .gitignore [hunk 2]
```diff
@@ -158,3 +158,6 @@ cython_debug/
 #  and can be added to the global gitignore or merged into this file.  For a more nuclear
 #  option (not recommended) you can uncomment the following to ignore the entire idea folder.
 #.idea/
+
+.DS_Store
+fmatch/main.py
```
Comment (jtaleric) [inline .gitignore:163]: Is main.py a local development tool?

Comment (shashank-boyapally) [reply .gitignore:163]: Yes Joe, it was intended to test the functions in my local.

File: fmatch/matcher.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
+from elasticsearch7 import Elasticsearch
+from elasticsearch.exceptions import NotFoundError
+import pandas as pd
+
```
Comment (jtaleric) [inline fmatch/matcher.py:4]: lets add a logging lib to help the user... some of the commented prints could be `logger.debug`  messages and we default the logger to info. thoughts?

Comment (shashank-boyapally) [reply fmatch/matcher.py:4]: Yes adding logger should be helpful if we move towards CLI.

**Review by jtaleric — APPROVED** [review]
lgtm - good start!

**Comment by paigerube14** [comment]
Agree good start! 
I think if we can try to parameterize burner_cpu_results some to be able to maybe read in a file instead of having it set in the script. Can be a future enhancement though!

**Review by paigerube14 — APPROVED** [review]

**Comment by shashank-boyapally** [comment]
We were planning to make it a CLI, so we should be able to put in a config in the future enhancements. Thank you for the review @jtaleric @paigerube14 . Merging!!

**Pull request merged**